alf/optimizers/optimizers.py

```diff
@@ -482,6 +482,22 @@ def add_param_group(self, param_group):
         capacity_ratio = self._capacity_ratio()
         self._adjust_capacity(capacity_ratio, {})
 
+    @common.add_method(NewCls)
+    def load_state_dict(self, state_dict):
+        """
+        Byte type is required for ``rng_state`` but within optimizer.load_state_dict,
+        it is converted to float.
+        This function first call the parent's ``load_state_dict()`` function, and
+        then make sure the ``rng_state`` is in correct data type.
+        """
+        super(NewCls, self).load_state_dict(state_dict)
+        if len(self.state) > 0:
+            for param_group in self.param_groups:
+                for p in param_group['params']:
+                    state = self.state[p]
+                    if 'rng_state' in state:
+                        state['rng_state'] = state['rng_state'].cpu().byte()
```

Comment:
Is rng_state always in cpu?

Reply:
Handled it in a more general way by using ``_rng_state_device``